### PR TITLE
chore: migration docs, schematics, and fix order of constructor deps for PlaceOrderComponent

### LINKED ISSUE #9881: Missing migration for `PlaceOrderComponent`
Constructor changed and variable `placeOrderSubscription` removed

image

### PR DESCRIPTION
**Pull request by Platonn**
Automatic migration can't inject constructor param at the start. It will always be at the end. To fix the automatic migration we change order of parameters in a component constructor.

fix #9881

### CODE REVIEW

File: projects/schematics/src/migrations/3_0/constructor-deprecations/data/place-order.component.migration.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  PLACE_ORDER_COMPONENT,
+  ROUTING_SERVICE,
+  SPARTACUS_CORE,
+  SPARTACUS_STOREFRONTLIB,
+  VIEW_CONTAINER_REF,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const PLACE_ORDER_COMPONENT_MIGRATION: ConstructorDeprecation = {
+  // projects\storefrontlib\src\cms-components\checkout\components\place-order\place-order.component.ts
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/3_0/constructor-deprecations/data/place-order.component.migration.ts]: let's change the black slashes to forward slashes

File: projects/schematics/src/migrations/3_0/component-deprecations/data/place-order.component.migration.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { PLACE_ORDER_COMPONENT } from '../../../../shared/constants';
+import { ComponentData } from '../../../../shared/utils/file-utils';
+
+export const PLACE_ORDER_COMPONENT_MIGRATION: ComponentData = {
+  selector: 'cx-place-order',
+  componentClassName: PLACE_ORDER_COMPONENT,
+  removedProperties: [
+    {
+      name: 'placeOrderSubscription',
+      comment: `'placeOrderSubscription' property was removed - no replacement`,
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/3_0/component-deprecations/data/place-order.component.migration.ts]: The name was changed to a more generic name `subscription` and just instantiates a new Subscription object to support other Observables than the placeOrder one

So I don't think it was removed, and maybe replaced? what do you think

File: projects/schematics/src/migrations/3_0/component-deprecations/data/place-order.component.migration.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { PLACE_ORDER_COMPONENT } from '../../../../shared/constants';
+import { ComponentData } from '../../../../shared/utils/file-utils';
+
+export const PLACE_ORDER_COMPONENT_MIGRATION: ComponentData = {
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/3_0/component-deprecations/data/place-order.component.migration.ts:4]: let's add path to file as comment here as well

File: projects/schematics/src/migrations/3_0/constructor-deprecations/data/place-order.component.migration.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  class: PLACE_ORDER_COMPONENT,
+  importPath: SPARTACUS_STOREFRONTLIB,
+  deprecatedParams: [
+    {
+      className: CHECKOUT_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: ROUTING_SERVICE,
+      importPath: ANGULAR_ROUTER,
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/3_0/constructor-deprecations/data/place-order.component.migration.ts]: should be `SPARTACUS_CORE`

**Review by bgambocjaviniar — DISMISSED** [review]
a few minor changes needed

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Review by Matejk00 — APPROVED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**